iceoryx_posh/include/iceoryx_posh/internal/gateway/gateway_generic.inl

```diff
@@ -78,8 +79,9 @@ GatewayGeneric<channel_t, gateway_t>::addChannel(const capro::ServiceDescription
                                                  const IceoryxPubSubOptions& options) noexcept
 {
     // Filter out wildcard services
-    if (service.getServiceID() == capro::AnyService || service.getInstanceID() == capro::AnyInstance
-        || service.getEventID() == capro::AnyEvent)
+    if (service.getServiceIDString() == capro::IdString_t(cxx::TruncateToCapacity, capro::AnyServiceString)
```

Comment:
This logic should be part of the service description. Could you please create a method called `doesContainWildcards()` and call this method here. Maybe you could come up with an even better name...

Reply:
I came to the conclusion that we need to differentiate between searching in the `ServiceRegistry` and creating `ServiceDescription`'s. Creating a `ServiceDescription` with a wildcard does not make sense semantically from my point of view. How can a `ServiceDescription` represent all of them with a wildcard? And even worse if you compare them, the equality operator would state them to be equal! They are not. They should always be well defined. This is why I removed them from `ServiceDescription` and added a possibility to search for Wildcard in #854. We can adapt the DDS gateway in a follow-up PR. Let's move this discussion over to #854. 

Also, see [this comment](https://github.com/eclipse-iceoryx/iceoryx/pull/852#discussion_r652853192) from @MatthiasKillat